tooltester/src/main/java/io/dockstore/tooltester/runWorkflow/WorkflowRunner.java

```diff
@@ -0,0 +1,250 @@
+        if (timesForEachTask == null) {
+            setTimeForEachTask();
+        }
+        Long totalTimeTaken = 0L;
+        for (TimeStatisticForOneTask time: timesForEachTask) {
+            totalTimeTaken += time.getTimeTakenInMilliseconds();
+        }
+        return totalTimeTaken;
+    }
+
+
+    private void printTimeStatistic() {
+        if (timesForEachTask == null) {
+            setTimeForEachTask();
+        }
+        for (TimeStatisticForOneTask time: timesForEachTask) {
+            Long timeTakenInMilliseconds = time.getTimeTakenInMilliseconds();
+            Long milliseconds = timeTakenInMilliseconds % MILLISECONDS_IN_SECOND;
+            Long seconds = timeTakenInMilliseconds / MILLISECONDS_IN_SECOND;
+            Long minutes = seconds / SECONDS_IN_MINUTE;
+            seconds = seconds % SECONDS_IN_MINUTE;
+
+            out("");
+            out("TASK NAME: " + time.getTaskName());
+            out("START TIME: " + time.getStartTime().toString());
+            out("END TIME: " + time.getEndTime().toString());
+            out("DURATION: " + minutes + " minutes " + seconds + " seconds " + milliseconds + " milliseconds");
+        }
+        out("");
+        Long totalTimeTakenMilliseconds = getTotalTimeInMilliseconds();
+        Long milliseconds = totalTimeTakenMilliseconds % MILLISECONDS_IN_SECOND;
+        Long seconds = totalTimeTakenMilliseconds / MILLISECONDS_IN_SECOND;
+        Long minutes = seconds / SECONDS_IN_MINUTE;
+        seconds = seconds % SECONDS_IN_MINUTE;
+        out("TOTAL TIME: " + minutes + " minutes " + seconds + " seconds " + milliseconds + " milliseconds");
+    }
+
+    public void printRunStatistics() {
+        if (isWorkflowFinished()) {
+            out("RUN STATISTICS:");
+            out("ENTRY NAME: " + getCompleteEntryName());
+            out("END STATE: " + state);
+            if (state.equals("COMPLETE")) {
+                printTimeStatistic();
+            }
+        } else {
+            errorMessage("Workflow is not finished, statistic are not available yet", COMMAND_ERROR);
+        }
+    }
+
+    private String getTotalTimeInISO861Standard() {
+        if (timesForEachTask == null) {
+            setTimeForEachTask();
+        }
+        Long totalTimeMilliseconds = getTotalTimeInMilliseconds();
+        Duration duration = Duration.ofMillis(totalTimeMilliseconds);
+        return duration.toString();
+    }
+
+    private Execution.ExecutionStatusEnum getExecutionStatus() {
+        if ("COMPLETE".equals(state)) {
+            return Execution.ExecutionStatusEnum.SUCCESSFUL;
+        } else {
+            return Execution.ExecutionStatusEnum.FAILED_RUNTIME_INVALID;
+        }
+    }
+
+    public void uploadRunInfo() {
+        List<Execution> executions = new ArrayList<>();
+        Execution execution = new Execution();
+        execution.setExecutionStatus(getExecutionStatus());
+
+        // Only include the time metric if the workflow successfully ran
```

Comment:
Interesting, the webservice doesn't specify that `ExecutionTime` should only be specified if it's successful. @denis-yuen any thoughts on whether it should?

Reply:
Time to failure is useful too, I'd include it.

Reply:
@fhembroff in this case, I would set the execution time here regardless of execution status

Reply:
One of the issues with this is that in the event of failure, AGC just reports the end time as `null`.

Would the desirable outcome in the event of a single task failing be to determine the amount of time it took for all the other tasks to run. This may not be entirely accurate, as the failed one may've run for a bit, without me knowing, as it would just report it's end time as `null`.

Reply:
@kathy-t it now sets the execution time regardless of execution status. Thanks for the feedback.